### PR TITLE
added all functionalities

### PR DESCRIPTION
**Pull request by balgyn98**
Bookstore simple java application

### CODE REVIEW

File: src/Book.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import java.util.Date;
+
+public class Book {
+
+    // fields
```
Comment (Alexmansar) [inline src/Book.java:5]: That's an unnecessary comment.

File: src/Book.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    private int id;
+    private String name;
+    private String author;
+    private Date publishedYear;
+    private int price;
+
+    // constructor
```
Comment (Alexmansar) [inline src/Book.java:12]: same

File: src/Main.java [hunk 3]
```diff
@@ -1,5 +1,167 @@
...
+                    System.out.println("User registration:");
+                    System.out.println("Please type your name: ");
+                    String userName = sc.nextLine();
+                    System.out.println("name = " + userName);
+                    System.out.println("Please type your email: ");
+                    String email = sc.nextLine();
+                    Customer customer = new Customer(userName, email, 1000);
+                    store.addCustomer(customer);
+                    System.out.println("You successfully registered");
+                    break;
```
Comment (Alexmansar) [inline src/Main.java:44]: use new switch style with ->

File: src/Main.java [hunk 4]
```diff
@@ -1,5 +1,167 @@
...
+                                break;
+                            }
+                        }
+
+                    } else {
+                        System.out.println("User does not exist");
+                    }
+                    break;
+
+                // get all orders
```
Comment (Alexmansar) [inline src/Main.java:91]: overall complex logic.
Please try to optimize it, at least create simple methods to make it easier to read.

File: src/Main.java [hunk 5]
```diff
@@ -1,5 +1,167 @@
...
+
 public class Main {
-    public static void main(String[] args) {
-        System.out.println("feature brench");
+    public static void main(String[] args) throws InterruptedException {
+        BookStore store = new BookStore();
+        Library library = new Library();
+        createLibrary(library);
+
+        Scanner sc = new Scanner(System.in);
```
Comment (Alexmansar) [inline src/Main.java:11]: scanner (full name, more easy to read)

File: src/Main.java [hunk 6]
```diff
@@ -1,5 +1,167 @@
...
+        Book book2 = new Book(2, "Pride and prejudice", "J.Rowling", date, 2100);
+        Book book3 = new Book(3, "Alchemist", "J.Rowling", date, 1800);
+        Book book4 = new Book(4, "Little women", "J.Rowling", date, 2500);
+        Book book5 = new Book(5, "The picture of Dorian Gray", "J.Rowling", date, 1200);
+
+        library.addBook(book1);
+        library.addBook(book2);
+        library.addBook(book3);
+        library.addBook(book4);
+        library.addBook(book5);
```
Comment (Alexmansar) [inline src/Main.java:132]: just to improve memory usage 
library.addBook(new Book(1, "Harry Potter", "J.Rowling", date, 1300));
i mean without creating variables, you we don't use it again)

File: src/Main.java [hunk 7]
```diff
@@ -1,5 +1,167 @@
...
+
+
+    public static List<Book> findBook(Customer user, Scanner sc, Library library,  List<Book> books){
+        int bookId = 0;
+
+        try {
+            System.out.println("Type book id to place the order:");
+            bookId = sc.nextInt();
+        } catch (InputMismatchException e){
+            sc.next();
```
Comment (Alexmansar) [inline src/Main.java:143]: not need here. you already in catch block

File: src/Main.java [hunk 8]
```diff
@@ -1,5 +1,167 @@
...
+
+        return books;
+    }
+
+    public static boolean isLessThanTwoMinutes(Date orderStartTime) {
+        Instant startInstant = orderStartTime.toInstant();
+        Instant nowInstant = Instant.now();
+
+        Duration duration = Duration.between(startInstant, nowInstant);
+        return duration.toMinutes() < 2;
```
Comment (Alexmansar) [inline src/Main.java:165]: what is 2? could you add please some name, or use like a constant?

File: src/Order.java [hunk 9]
```diff
@@ -0,0 +1,116 @@
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class Order {
+    private static final AtomicInteger count = new AtomicInteger(0);
+    private int ID;
```
Comment (Alexmansar) [inline src/Order.java:8]: small style
just id;

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@balgyn98 Hi)) In general good job, but I have some comments) so, please just fix it and send me again)

File: src/Main.java [hunk 10]
```diff
@@ -1,5 +1,149 @@
...
+
+    }
+
+    public static Customer login(Scanner scanner){
+        System.out.println("type the user email");
+        String username = scanner.nextLine();
+        Customer user = store.getCustomerByName(username);
+
+        return user;
+    }
```
Comment (Alexmansar) [inline src/Main.java:68]: you can create CustomerService and more logic

File: src/Main.java [hunk 11]
```diff
@@ -1,5 +1,149 @@
...
+
+    public static boolean isSessionLessThanTwoMinutes(Date orderStartTime) {
+        Instant startInstant = orderStartTime.toInstant();
+        Instant nowInstant = Instant.now();
+
+        Duration duration = Duration.between(startInstant, nowInstant);
+        return duration.toMinutes() < maxSessionDuration;
+    }
+
+    public static boolean createUser(Scanner scanner){
```
Comment (Alexmansar) [inline src/Main.java:87]: same for User

File: src/Main.java [hunk 12]
```diff
@@ -1,5 +1,149 @@
...
+            store.addCustomer(customer);
+            System.out.println("You successfully registered");
+        } catch (Exception e){
+            System.out.println(e.getMessage());
+            return false;
+        }
+        return true;
+    }
+
+    public static void createOrder( Customer user, Scanner scanner){
```
Comment (Alexmansar) [inline src/Main.java:103]: and for rorder

In general, move such methods to the specific classes))

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@balgyn98 I have new general comments) please, check it

File: src/service/OrderService.java [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+import java.util.Date;
+import java.util.List;
+import java.util.Scanner;
+
+
+public class OrderService {
+    static BookStoreService bookStoreService;
+    final static int maxSessionDuration = 2; //session should be no longer than 2 minutes
+
+    public static void createOrder(Customer user, Scanner scanner, Library library, BookStore store){
```
Comment (Alexmansar) [inline src/service/OrderService.java:17]: same

File: src/Main.java [hunk 14]
```diff
@@ -1,5 +1,77 @@
...
+
+    public static void createLibrary(Library library){
+        Date date = new Date();
+        library.addBook(new Book(1, "Harry Potter", "J.Rowling", date, 1300));
+        library.addBook(new Book(2, "Pride and prejudice", "J.Rowling", date, 2100));
+        library.addBook(new Book(3, "Alchemist", "J.Rowling", date, 1800));
+        library.addBook(new Book(4, "Little women", "J.Rowling", date, 2500));
+        library.addBook(new Book(5, "The picture of Dorian Gray", "J.Rowling", date, 1200));
     }
 }
```
Comment (Alexmansar) [inline src/Main.java:77]: and now, I think we can move it to the Library))

**Review by Alexmansar — APPROVED** [review]
@balgyn98 
Hi again))) good job in general)
please, fix last comment and merge it))

File: src/Main.java (hunk 14 above)
Comment (balgyn98) [reply src/Main.java:77]: thank you so much!) learnt many things from this task

**Pull request merged**